### PR TITLE
Update issue templates

### PR DESCRIPTION
**Pull request by romanlutz**
Setting up issue templates that people need to fill when reporting bugs or filing feature requests. This should help in getting all the relevant information from the start.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+name: Bug report
+about: Create a report to help us improve
+title: BUG
+labels: bug
```
Comment (adrinjalali) [inline .github/ISSUE_TEMPLATE/bug_report.md:5]: we found that it's useful to have the initial tag as "Bug: triage" or something, rather bug, and once the reported issue is triaged, a maintainer changes the label to "bug". Quite frequently reported "bugs" are not really bugs.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+assignees: ''
+
+---
+
+**Describe the bug**
+A clear and concise description of what the bug is.
```
Comment (adrinjalali) [inline .github/ISSUE_TEMPLATE/bug_report.md:11]: You can leave the description of what users should do in comments so that by default they're not rendered. You can see [here](https://raw.githubusercontent.com/scikit-learn/scikit-learn/master/PULL_REQUEST_TEMPLATE.md) an example, and the syntax looks like:

```
<!--
**Describe the bug**
A clear and concise description of what the bug is.
here comes the text which won't be rendered
-->
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+**Screenshots**
+If applicable, add screenshots to help explain your problem.
+
+**Desktop (please complete the following information):**
+ - OS: [e.g. iOS]
+ - Browser [e.g. chrome, safari]
+ - Fairlearn version [e.g. 0.4.4]
+ - Python version [e.g. 3.5.4]
+ - version of Fairlearn dependencies, i.e. ideally the output of `pip freeze`
```
Comment (adrinjalali) [inline .github/ISSUE_TEMPLATE/bug_report.md:27]: it's easier if you give people a way to extract this information. For instance, [we have](https://raw.githubusercontent.com/scikit-learn/scikit-learn/master/.github/ISSUE_TEMPLATE/bug_report.md):

```
Please run the following snippet and paste the output below.
For scikit-learn >= 0.20:
import sklearn; sklearn.show_versions()
For scikit-learn < 0.20:
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("NumPy", numpy.__version__)
import scipy; print("SciPy", scipy.__version__)
import sklearn; print("Scikit-Learn", sklearn.__version__)
import imblearn; print("Imbalanced-Learn", imblearn.__version__)
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+**Expected behavior**
+A clear and concise description of what you expected to happen.
+
+**Screenshots**
+If applicable, add screenshots to help explain your problem.
+
+**Desktop (please complete the following information):**
+ - OS: [e.g. iOS]
+ - Browser [e.g. chrome, safari]
+ - Fairlearn version [e.g. 0.4.4]
```
Comment (riedgar-ms) [inline .github/ISSUE_TEMPLATE/bug_report.md:25]: As we discovered with Abe, we should explicitly ask if they installed via pip or by checking out from GitHub.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+Steps to reproduce the behavior.
+
+**Expected behavior**
+A clear and concise description of what you expected to happen.
+
+**Screenshots**
+If applicable, add screenshots to help explain your problem.
+
+**Desktop (please complete the following information):**
+ - OS: [e.g. iOS]
```
Comment (riedgar-ms) [inline .github/ISSUE_TEMPLATE/bug_report.md:23]: iOS?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ - Browser [e.g. chrome, safari]
```
Comment (riedgar-ms) [inline .github/ISSUE_TEMPLATE/bug_report.md:24]: We should probably include Edge.... ;-)

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (romanlutz) [reply .github/ISSUE_TEMPLATE/bug_report.md:5]: Great suggestion! Will adjust...

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (romanlutz) [reply .github/ISSUE_TEMPLATE/bug_report.md:11]: I was wondering about that... would be annoying to have them displayed :-) Will address that...

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 5 above)
Comment (romanlutz) [reply .github/ISSUE_TEMPLATE/bug_report.md:23]: :D this was the default template. Looks like I missed this in adjusting it

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 6 above)
Comment (romanlutz) [reply .github/ISSUE_TEMPLATE/bug_report.md:24]: and Firefox!

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (romanlutz) [reply .github/ISSUE_TEMPLATE/bug_report.md:5]: The link to the sklearn templates was pure gold. I copied those into my directory and adjusted just a few things that were sklearn specific to be Fairlearn-specific. Please take another look :-)

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (romanlutz) [reply .github/ISSUE_TEMPLATE/bug_report.md:11]: This is great!

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 3 above)
Comment (romanlutz) [reply .github/ISSUE_TEMPLATE/bug_report.md:27]: wouldn't `pip freeze` be much more comprehensive? Sometimes it's something annoying like, say, pytest, or jupyter, or who knows what :-)

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 4 above)
Comment (romanlutz) [reply .github/ISSUE_TEMPLATE/bug_report.md:25]: I added a pip freeze instruction which would tell us how it's installed. Additionally, I've also asked for the Fairlearn version just to be sure.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 5 above)
Comment (romanlutz) [reply .github/ISSUE_TEMPLATE/bug_report.md:23]: this is now gone

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+<!-- If applicable, add screenshots to help explain your problem. -->
+
+#### Versions
+<!--
+Please provide the following information:
+- OS: [e.g. Windows]
+- Browser (if you're reporting a dashboard bug in jupyter): [e.g. Edge, Firefox, Chrome, Safari]
+- Python version: [e.g. 3.7.4]
+- Fairlearn version: [e.g. 0.4.5 or installed from master branch in editable mode]
+- version of Python packages: please run `pip freeze` and paste the output.
```
Comment (adrinjalali) [inline .github/ISSUE_TEMPLATE/bug_report.md:73]: The issue with `pip freeze` is that it doesn't list conda packages AFAIK. So you don't really get a good idea of what's installed. I find knowing the versions of just dependencies helps more.

In general, giving them a piece of code to run and paste the result lowers the barrier substantially.

Comment (riedgar-ms) [reply .github/ISSUE_TEMPLATE/bug_report.md:73]: Fair point. We do still need to know if they have installed Fairlearn from pip, or if they checked out the repo, though.

Comment (adrinjalali) [reply .github/ISSUE_TEMPLATE/bug_report.md:73]: fairlearn version should give you that. The repo should always have a `dev[n]` version.

Comment (riedgar-ms) [reply .github/ISSUE_TEMPLATE/bug_report.md:73]: That would be a change to how we're organising things right now. Currently, the repo has the last-released version. Which is not to say that we don't need to rethink our release process, as we've been mentioning in other discussions.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 3 above)
Comment (romanlutz) [reply .github/ISSUE_TEMPLATE/bug_report.md]: switched this to have show_versions

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 7 above)
Comment (romanlutz) [reply .github/ISSUE_TEMPLATE/bug_report.md]: moved away from pip freeze to show_versions. This `dev[n]` discussion is beyond the topic of the PR but I encourage having it anyway

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+<!-- If applicable, add screenshots to help explain your problem. -->
+
+#### Versions
+<!--
+Please provide the following information:
+- OS: [e.g. Windows]
+- Browser (if you're reporting a dashboard bug in jupyter): [e.g. Edge, Firefox, Chrome, Safari]
+- Python version: [e.g. 3.7.4]
+- Fairlearn version: [e.g. 0.4.5 or installed from master branch in editable mode]
+- version of Python packages: please run the following snipped and paste the output:
```
Comment (MiroDudik) [inline .github/ISSUE_TEMPLATE/bug_report.md:73]: snipped -> snippet

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+#### Describe the bug
+<!--
+A clear and concise description of what the bug is.
+-->
+
+#### Steps/Code to Reproduce
+<!--
+Please add a minimal example that we can reproduce the error by running the
+code. Be as succinct as possible, do not depend on external data. In short, we
```
Comment (MiroDudik) [inline .github/ISSUE_TEMPLATE/bug_report.md:32]: Improve wording. How about:
```
Please add a minimal example (in the form of code) that reproduces the error.
Be as succinct as possible, ...
```

File: fairlearn/utils/__init__.py [hunk 10]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (MiroDudik) [inline fairlearn/utils/__init__.py:1]: we don't need a new package for this! let's just move this into the top module, so we only have `fairlearn.show_versions`. See: https://scikit-learn.org/stable/modules/generated/sklearn.show_versions.html

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+name: Feature request
+about: Suggest an idea for this project
+title: ''
+labels: feature-request
+assignees: ''
+
+---
+
+<!--
+If your issue is a usage question, please submit it in one of these other
```
Comment (MiroDudik) [inline .github/ISSUE_TEMPLATE/feature_request.md:11]: Do we want to mention `fairlearn-proposals` here? I'm also good leaving as is for now. We can move the feature requests into `fairlearn-proposals` repo as needed.

File: fairlearn/utils/__init__.py (hunk 10 above)
Comment (romanlutz) [reply fairlearn/utils/__init__.py:1]: They have `utils`, but they hide that fact by making it available in the top level module https://github.com/scikit-learn/scikit-learn/blob/95d4f0841/sklearn/utils/_show_versions.py#L75
I don't particularly care, so I've moved it into fairlearn/

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 11 above)
Comment (romanlutz) [reply .github/ISSUE_TEMPLATE/feature_request.md:11]: I think people may get confused if we give them the option. Better to triage that ourselves.

Comment (MiroDudik) [reply .github/ISSUE_TEMPLATE/feature_request.md:11]: sounds good.
